src/thunderbots/software/util/canvas_messenger/canvas_messenger.cpp

```diff
@@ -0,0 +1,113 @@
+#include "canvas_messenger.h"
+
+#include "util/constants.h"
+
+namespace Util
+{
+    std::shared_ptr<CanvasMessenger> CanvasMessenger::getInstance()
+    {
+        static std::shared_ptr<CanvasMessenger> vm(new CanvasMessenger);
+        return vm;
+    }
+
+    void CanvasMessenger::initializePublisher(ros::NodeHandle node_handle)
+    {
+        this->publisher = node_handle.advertise<CanvasLayer>(
+            Util::Constants::VISUALIZER_DRAW_LAYER_TOPIC, 8);
```

Comment:
`8` should probably be a constant declared `private` to this class

Reply:
Added the constant